### DISCUSSION TITLE
plugins with additional dependency development

### DISCUSSION THREAD

**Discussion by harin** [Budibase/budibase#10183] [Help] [answered]
I am trying to develop a `mssql` plugin that allows the option `trustServerCertificate` to be set. The [tutorial ](https://docs.budibase.com/docs/custom-datasource) works fine until I added `mssql` as a dependency by doing `yarn add mssql`.

After that, trying to add the datasource would give me the following error.
> Cannot find module 'mssql' Require stack: - /tmp/.budibase/datasource/private-mssql - /app/dist/utilities/fileSystem/plugin.js - /app/dist/utilities/fileSystem/index.js - /app/dist/index.js

Note: I am testing this through `budi plugins --dev` and going to `localhost:10000`

**Comment by ConorWebb96** [comment, marked as answer]
Hey @harin,

I think this may be better if you add this functionality into the core product. You could fork the Budibase repo, implement the enhancement and then create a pull request to get this added to the product itself. (I think this would be pretty straight forward to implement looking at other connectors that have it.)

**Postgres example:**
```
ssl: {
      type: DatasourceFieldType.BOOLEAN,
      default: false,
      required: false,
    },
```
Note that the Default setting should be false for this trustServerCert.

For the `microsoftSqlServer.ts` you likely would only need to add something like the following to it.

```
constructor(config: MSSQLConfig) {
    super(SqlClient.MS_SQL)
    this.config = config
    const clientCfg = {
      ...this.config,
      options: {
        encrypt: this.config.encrypt,
        enableArithAbort: true,
        trustServerCertificate: this.config.trustServerCertificate // add something like this
      },
    }
```

If you are set on creating a custom component for this. Please try the following:

- Make sure that mssql is installed as a dependency in your project. You can install it by running the following command in your project directory:

`yarn add mssql`

- Check that there are no typos or errors in the path to the module in your import statement. The error message you posted suggests that the error may be occurring in the private-mssql file, so you should double-check the path in that file.

I hope this helps!

**Comment by harin** [comment]
Thank you for the suggestion, I didn't add this to the core product because I didn't have time to understand how it works or how to test it, plus there's a nice tutorial on getting started with plugins. I see what I can do once I have the time.
